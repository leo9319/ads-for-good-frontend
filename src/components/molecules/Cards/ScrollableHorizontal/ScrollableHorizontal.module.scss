@import '../../../../styles/colours';

.container {
  display: grid;
  width: 100%;
  gap: var(--spacing-lg);
  background-color: $color-neutral-50;

  .header {
    display: flex;
    gap: var(--spacing-xs);
    justify-content: space-between;

    .contentHeader {
      gap: var(--spacing-2xs);

      .heading {
        font-weight: 600;
        color: $color-neutral-950;
        font-size: var(--typography-font-size-7);
        line-height: var(--typography-line-height-7);

        @media (max-width: 360px) {
          font-size: var(--typography-font-size-5);
          line-height: var(--typography-line-height-5);
        }
      }

      .description {
        color: $color-neutral-600;
        font-size: var(--typography-font-size-4);
        line-height: var(--typography-line-height-4);

        @media (max-width: 360px) {
          font-size: var(--typography-font-size-3);
          line-height: var(--typography-line-height-3);
        }
      }
    }

    .navButtons {
      align-items: center;
      gap: var(--spacing-2xs);

      .btn {
        width: var(--spacing-xl) !important;
        height: var(--spacing-xl) !important;
      }
    }
  }

  .body {
    .contentContainer {
      gap: var(--spacing-sm);
      margin-bottom: var(--spacing-lg);
      padding-bottom: var(--spacing-xs);
    }
  }

  &.inverse {
    background-color: $color-white;
  }
}
