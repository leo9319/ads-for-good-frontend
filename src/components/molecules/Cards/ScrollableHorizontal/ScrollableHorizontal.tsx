import React, {
  ReactElement,
  ReactNode,
  useEffect,
  useRef,
  useState,
} from 'react';
import Box from '@radix-styles/atoms/Box';
import Flex from '@radix-styles/atoms/Flex';
import Text from '@radix-styles/atoms/Text';
import Button from '@radix-styles/atoms/Button';
import ScrollArea from '@radix-styles/atoms/ScrollArea';
import { classNames } from '@utils/common/classNames';
import { AnalyticsHandler } from '@utils/Analytics';

import style from './ScrollableHorizontal.module.scss';

/**
 * @interface
 * The Scrollable Horizontal Card props type of {@link ScrollableHorizontal}
 */
export interface ScrollableHorizontalProps {
  /**
   * Length of items
   */
  itemsLength: number;

  /**
   * Heading/title displayed above or within the list.
   */
  heading: string;

  /**
   * Determines whether to show the description.
   * If `true`, the `description` will be visible.
   */
  showDescription?: boolean;

  /**
   * Description text for the list.
   * Used when `showDescription` is `true`.
   */
  description?: string;

  /**
   * Defines the visual mode/theme.
   */
  mode?: 'primary' | 'inverse';

  /**
   * Optional CSS class names for additional styling.
   */
  classname?: string;
  /**
   * Children as jsx, react component will show inside the modal
   * @property
   */
  children: ReactNode;
  /**
   * Unique name to assign and support the scrolling behaviour
   * @property
   */
  name: string;
}

/**
 *
 * A horizontally scrollable list that displays a set of items with optional descriptions.
 * It includes support for different sizes and modes, along with configurable styling.
 *
 * @group Molecules
 * @category Component
 *
 * @param {ScrollableHorizontal} props - The Scrollable Horizontal props
 * @returns {ReactElement} - Scrollable HorizontalComponent
 *
 */
export const ScrollableHorizontal = ({
  itemsLength,
  heading,
  description,
  classname,
  mode = 'primary',
  showDescription = true,
  children,
  name,
}: ScrollableHorizontalProps): ReactElement => {
  const [active, setActive] = useState<number>(1);
  const [hideBtn, setHideBtn] = useState<boolean>(false);
  const scrollContainerRef = useRef<HTMLDivElement>(null);
  const { trackClick } = AnalyticsHandler();

  // Calculating the ignorable indices
  const calcIgnorableIndex = () => {
    const containerWidth = scrollContainerRef?.current?.clientWidth;
    if (containerWidth) {
      const firstChildElement = scrollContainerRef?.current?.querySelector(
        `#${name}-1`
      );
      const childWidth = firstChildElement?.clientWidth;
      if (childWidth) {
        const gapBetweenCards = 12;

        const cardCount = Math.floor(containerWidth / childWidth);
        const remainingWidth = containerWidth % childWidth;

        const totalGapWidth = cardCount * gapBetweenCards;

        if (remainingWidth > totalGapWidth) {
          const usableWidth = containerWidth - totalGapWidth;
          const visibleCardCount = Math.floor(usableWidth / childWidth);

          return visibleCardCount;
        }
        const reducedGapWidth = (cardCount - 1) * gapBetweenCards;
        const usableWidth = containerWidth - reducedGapWidth;
        const visibleCardCount = Math.floor(usableWidth / childWidth);

        return visibleCardCount;
      }
    }

    return 0;
  };

  // The function to get the next value
  const next = () => {
    const ignorableIndex = calcIgnorableIndex();
    const activeIndex =
      active === itemsLength - (ignorableIndex - 1) || active === itemsLength
        ? 1
        : active + 1;
    return activeIndex;
  };

  // The function to get the previous value
  const previous = () => {
    const ignorableIndex = calcIgnorableIndex();
    if (active === 1) {
      return itemsLength;
    } else if (active === itemsLength) {
      return itemsLength - ignorableIndex;
    }
    return active - 1;
  };

  // Handler Function of Next button
  const handleNext = () => {
    const newIndex = next();
    setActive(() => newIndex);
    trackClick({
      target: '',
      location: 'horizontal scroller',
      name: heading,
      type: 'button',
      contentTypeOrPosition: 'next',
    });
  };

  // Handler Function of Previous button
  const handlePrevious = () => {
    const newIndex = previous();
    setActive(() => newIndex);
    trackClick({
      target: '',
      location: 'horizontal scroller',
      name: heading,
      type: 'button',
      contentTypeOrPosition: 'previous',
    });
  };

  // Scrolling the current view into target view
  const scrollToActive = (index: number) => {
    // Scroll to the target card inside the container
    document.getElementById(`${name}-${index}`)?.scrollIntoView({
      behavior: 'smooth',
      block: 'nearest',
      inline: 'start',
    });
  };

  // Triggering the scrollToActive when on Mounting and after Updating the active
  useEffect(() => {
    scrollToActive(active);
  }, [active]);

  //To hide the prev and next button
  useEffect(() => {
    setTimeout(() => {
      if (scrollContainerRef?.current) {
        const elements = scrollContainerRef?.current.children;
        for (const element of elements) {
          if (element.getAttribute('data-state')) {
            setHideBtn(true);
          }
        }
      }
    }, 250);
  });

  // Enabling ResizeObserver to update the view to the initial position
  useEffect(() => {
    const resizeObserver = new ResizeObserver(() => {
      setActive(1);
    });

    if (scrollContainerRef.current) {
      resizeObserver.observe(scrollContainerRef.current);
    }

    return () => {
      if (scrollContainerRef.current) {
        resizeObserver.unobserve(scrollContainerRef.current);
      }
    };
  }, []);

  // Concatinating the size, mode and additional class name
  const concatinatedClasses = classNames(
    style.container,
    classname,
    style[mode]
  );

  // Defining button mode
  const buttonMode = mode === 'primary' ? 'inverse' : 'secondary';

  return (
    <Box className={concatinatedClasses}>
      <Box className={style.header}>
        <Flex direction="column" className={style.contentHeader}>
          <Text as="p" className={style.heading}>
            {heading}
          </Text>
          {showDescription && description && (
            <Text as="p" className={style.description}>
              {description}
            </Text>
          )}
        </Flex>
        {hideBtn && (
          <Flex className={classNames(style.navButtons)}>
            <Button
              text=""
              rounded
              size="md"
              icon="icon-previous"
              ariaLabelName="Previous Button"
              mode={buttonMode}
              className={style.btn}
              onClick={handlePrevious}
            />
            <Button
              text=""
              rounded
              size="md"
              icon="icon-next"
              ariaLabelName="Next Button"
              mode={buttonMode}
              className={style.btn}
              onClick={handleNext}
            />
          </Flex>
        )}
      </Box>
      <ScrollArea classname={style.body} ref={scrollContainerRef}>
        <Flex className={style.contentContainer}>{children}</Flex>
      </ScrollArea>
    </Box>
  );
};

export default ScrollableHorizontal;
