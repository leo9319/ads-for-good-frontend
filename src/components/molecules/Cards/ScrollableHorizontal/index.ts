export { default } from './ScrollableHorizontal';
export { ScrollableHorizontal } from './ScrollableHorizontal';
export type { ScrollableHorizontalProps } from './ScrollableHorizontal';
