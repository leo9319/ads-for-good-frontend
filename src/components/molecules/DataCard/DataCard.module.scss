@import '../../../styles/colours';

@mixin apply-size($fontSize, $fontWeight, $lineHeight, $lineClamp: 1) {
  font-size: var($fontSize);
  font-weight: $fontWeight;
  line-height: var($lineHeight);
  line-clamp: $lineClamp;
  -webkit-line-clamp: $lineClamp;
}

@mixin apply-line-clamp {
  display: -webkit-box;
  overflow: hidden;
  align-self: stretch;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
}

.dataCardContainer {
  display: flex;
  width: 160px;
  height: 213px;
  flex-direction: column;
  align-items: flex-start;
  padding: var(--spacing-md);
  justify-content: space-between;
  background-color: $color-white;
  box-shadow: 0 1px 3px 0 $color-box-shadow;
  font-family: var(--typography-default-font-family);

  --card-border-width: 0px;
  --card-border-radius: var(--spacing-md);
  --card-background-color: unset;

  &.surface {
    border: 1px solid $color-neutral-100;
  }

  .headerContainer {
    width: 100%;
    display: grid;
    gap: var(--spacing-xs);

    .header {
      display: grid;
      gap: var(--spacing-2xs);

      .heading {
        @include apply-size(
          --typography-font-size-5,
          700,
          --typography-line-height-5
        );
        @include apply-line-clamp();

        color: $color-neutral-950;
      }

      .subHeading {
        @include apply-size(
          --typography-font-size-2,
          600,
          --typography-line-height-2,
          2
        );
        @include apply-line-clamp();

        color: $color-neutral-950;
      }
    }

    .content {
      @include apply-size(
        --typography-font-size-1,
        400,
        --typography-line-height-1,
        5
      );
      @include apply-line-clamp();

      color: $color-neutral-600;
    }
  }

  .footer {
    width: 100%;
  }

  &.size {
    &-2 {
      width: 248px;
      height: 331px;
      padding: var(--spacing-2xl);
      --card-border-radius: var(--spacing-xl);

      .headerContainer {
        .header {
          .heading {
            @include apply-size(
              --typography-font-size-8,
              600,
              --typography-line-height-8
            );
          }

          .subHeading {
            @include apply-size(
              --typography-font-size-4,
              600,
              --typography-line-height-4,
              2
            );
          }
        }

        .content {
          @include apply-size(
            --typography-font-size-3,
            400,
            --typography-line-height-3,
            7
          );
        }
      }
    }

    &-3 {
      width: 348px;
      height: 464px;
      padding: var(--spacing-2xl);
      --card-border-radius: var(--spacing-xl);

      .headerContainer {
        gap: var(--spacing-md);

        .header {
          .heading {
            @include apply-size(
              --typography-font-size-10,
              600,
              --typography-line-height-10
            );
          }

          .subHeading {
            @include apply-size(
              --typography-font-size-4,
              600,
              --typography-line-height-4,
              2
            );
          }
        }

        .content {
          @include apply-size(
            --typography-font-size-4,
            400,
            --typography-line-height-4,
            10
          );
        }
      }
    }
  }
}
