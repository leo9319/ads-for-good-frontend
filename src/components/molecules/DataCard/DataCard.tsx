import React, { ReactElement } from 'react';
import Box from '@radix-styles/atoms/Box';
import Card from '@radix-styles/atoms/Card';
import Tag from '@radix-styles/atoms/Tag';
import Text from '@radix-styles/atoms/Text';
import { classNames } from '@utils/common/classNames';
import educationLogo from '@assets/icons/emblemLogo/Education.svg';
import foodLogo from '@assets/icons/emblemLogo/Food.svg';
import healthLogo from '@assets/icons/emblemLogo/Health.svg';
import waterLogo from '@assets/icons/emblemLogo/Water.svg';
import markLogo from '@assets/icons/emblemLogo/Mark.svg';
import childProtection from '@assets/icons/emblemLogo/Child_Protection.svg';

import style from './DataCard.module.scss';

/**
 * @interface
 * The Data Card props type of {@link DataCard}
 */
export interface DataCardProps {
  /**
   * Determines the size of the Data Card
   * @defaultValue `1`
   * @property
   */
  size?: '1' | '2' | '3';
  /**
   * Controls the visual style of the Data Card
   * @defaultValue `default`
   * @property
   */
  variant?: 'default' | 'surface';
  /**
   * Text displayed next to the Logo
   * @property
   */
  logoText: string;
  /**
   * Name of the Logo to be displayed
   * @property
   */
  logoName: string;
  /**
   * Main title or heading of the Card
   * @property
   */
  heading: string;
  /**
   * Subtitle or secondary heading of the Card
   * @property
   */
  subHeading: string;
  /**
   * A brief summary or description of the Card's content
   * @property
   */
  content: string;
  /**
   * Additional class names for customizing styles
   * @property
   */
  containerClass?: string;
}

/**
 * @interface
 * The Logo object types
 */
interface ILogoObj {
  [key: string]: string;
}

/**
 * The Data Card is a structured component that organizes related content and actions.
 * It features a heading, subheading, and content, while also supporting customizable
 * sizes, visual variants, and a logo.
 *
 * @group Molecules
 * @category Component
 *
 * @param {DataCardProps} props - The Data Card props
 * @returns {ReactElement} - Data Card Component
 *
 */
export const DataCard = ({
  heading,
  subHeading,
  content,
  logoName,
  logoText,
  size = '1',
  variant = 'default',
  containerClass = '',
}: DataCardProps): ReactElement => {
  // Temporary object to obtain the path of the emblem logo
  const logoObj: ILogoObj = {
    education: educationLogo,
    food: foodLogo,
    health: healthLogo,
    water: waterLogo,
    mark: markLogo,
    child: childProtection,
  };

  // Function to add the prefix with card size
  const getSize = (size: string): string => `size-${size}`;

  const logoPath: string = logoObj[logoName];

  const newClass = classNames(
    style.dataCardContainer,
    containerClass,
    style[getSize(size)],
    style[variant]
  );

  return (
    <Card className={newClass}>
      <Box className={style.headerContainer}>
        <Box className={style.header}>
          <Text as="p" className={style.heading}>
            {heading}
          </Text>
          <Text as="p" className={style.subHeading}>
            {subHeading}
          </Text>
        </Box>
        <Text as="p" className={style.content}>
          {content}
        </Text>
      </Box>
      <Box className={style.footer}>
        <Tag showAvatar mode="default" text={logoText} iconUrl={logoPath} />
      </Box>
    </Card>
  );
};

export default DataCard;
