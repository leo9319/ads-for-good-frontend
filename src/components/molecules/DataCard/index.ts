export type { DataCardProps } from './DataCard';
export { default } from './DataCard';
export { DataCard } from './DataCard';
