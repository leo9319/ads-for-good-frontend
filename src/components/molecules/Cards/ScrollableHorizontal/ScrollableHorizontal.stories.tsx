import React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import ScrollableHorizontal from './ScrollableHorizontal';
import Theme from '@radix-styles/atoms/Theme';
import DataCard, { DataCardProps } from '@components/molecules/DataCard';
import Box from '@radix-styles/atoms/Box';

const DataCardList: DataCardProps[] = [
  {
    heading: '1M+',
    subHeading: 'People served',
    content:
      'Our livelihood improvement programs enable millions globally to reach better outcomes',
    logoName: 'education',
    logoText: 'Education',
  },
  {
    heading: '2M+',
    subHeading: 'Supported youth',
    content: 'Youth gained safe and supportive learning environments',
    logoName: 'food',
    logoText: 'Food',
  },
  {
    heading: '3M+',
    subHeading: 'People served',
    content:
      'Our livelihood improvement programs enable millions globally to reach better outcomes',
    logoName: 'water',
    logoText: 'Water',
  },
  {
    heading: '4M+',
    subHeading: 'Children protected',
    content:
      'WVC works to protect children through rights and safety initiatives',
    logoName: 'health',
    logoText: 'Health',
  },
  {
    heading: '5M+',
    subHeading: 'Gained water access',
    content: 'WVC helped people gained access to clean water.',
    logoName: 'education',
    logoText: 'Education',
  },
  {
    heading: '6M+',
    subHeading: 'People served',
    content:
      'Our livelihood improvement programs enable millions globally to reach better outcomes',
    logoName: 'food',
    logoText: 'Food',
  },
  {
    heading: '7M+',
    subHeading: 'People served',
    content:
      'Our livelihood improvement programs enable millions globally to reach better outcomes',
    logoName: 'water',
    logoText: 'Water',
  },
  {
    heading: '8M+',
    subHeading: 'Children protected',
    content:
      'WVC works to protect children through rights and safety initiatives',
    logoName: 'education',
    logoText: 'Education',
  },
  {
    heading: '9M+',
    subHeading: 'Supported youth',
    content: 'Youth gained safe and supportive learning environments',
    logoName: 'health',
    logoText: 'Health',
  },
  {
    heading: '10M+',
    subHeading: 'Gained water access',
    content: 'WVC helped people gained access to clean water.',
    logoName: 'water',
    logoText: 'Water',
  },
];

const meta = {
  title: 'Library/Molecules/Cards/ScrollableHorizontal',
  component: ScrollableHorizontal,
  decorators: [
    Story => (
      <Theme>
        <Story />
      </Theme>
    ),
  ],
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    itemsLength: {
      control: {
        type: 'number',
      },
    },
    heading: {
      control: 'text',
    },
    description: {
      control: 'text',
    },
    mode: {
      control: {
        type: 'select',
      },
      options: ['primary', 'inverse'],
    },
    showDescription: {
      control: 'boolean',
    },
    classname: {
      control: 'text',
    },
    children: {
      control: 'object',
    },
    name: {
      control: 'text',
    },
  },
  args: {
    showDescription: true,
    itemsLength: DataCardList.length,
    heading: 'Impact Summary',
    description:
      'Discover our global impact and collective support for communities.',
    name: 'child',
    children: (
      <>
        {DataCardList?.map(
          (
            { heading, subHeading, content, logoName, logoText, size, variant },
            idx
          ) => (
            <Box key={idx + '_scroll_card'} id={'child-' + (idx + 1)}>
              <DataCard
                size={size || '1'}
                variant={variant || 'surface'}
                content={content || ''}
                heading={heading || 'Heading'}
                logoName={logoName || 'food'}
                logoText={logoText || 'Food'}
                subHeading={subHeading || 'Sub Heading'}
              />
            </Box>
          )
        )}
      </>
    ),
  },
} satisfies Meta<typeof ScrollableHorizontal>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    mode: 'primary',
  },
};

export const Inverse: Story = {
  args: {
    mode: 'inverse',
  },
};
