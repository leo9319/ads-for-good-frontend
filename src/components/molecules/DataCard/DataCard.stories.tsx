import React from 'react';
import DataCard from './DataCard';
import type { Meta, StoryObj } from '@storybook/react';
import { Theme } from '@radix-ui/themes';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title: 'Library/Molecules/DataCard',
  component: DataCard,
  decorators: [
    Story => (
      <Theme>
        <Story />
      </Theme>
    ),
  ],
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes
  argTypes: {
    heading: { control: 'text' },
    subHeading: { control: 'text' },
    content: { control: 'text' },
    size: { control: 'select', options: ['1', '2', '3'] },
    variant: { control: 'select', options: ['default', 'surface'] },
    logoText: { control: 'text' },
    logoName: {
      control: 'select',
      options: ['education', 'food', 'health', 'water', 'mark', 'child'],
    },
    containerClass: { control: 'text' },
  },
  // more on args: https://storybook.js.org/docs/writing-stories/args#args-object
  args: {
    heading: '4M+',
    subHeading: 'People served',
    content:
      'Our livelihood improvement programs enable millions globally to reach better outcomes',
    size: '1',
    logoName: 'education',
  },
} satisfies Meta<typeof DataCard>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args

export const Default: Story = {
  args: {
    variant: 'default',
    logoText: 'Water',
    logoName: 'water',
  },
};

export const Surface: Story = {
  args: {
    variant: 'surface',
    logoText: 'Food',
    logoName: 'food',
  },
};
