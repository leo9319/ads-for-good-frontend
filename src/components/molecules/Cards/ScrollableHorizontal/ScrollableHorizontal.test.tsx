import React from 'react';
import { render } from '@testing-library/react';
import ScrollableHorizontal from './ScrollableHorizontal';
import DataCard from '@components/molecules/DataCard';

describe('Scrollable Horizontal Component', () => {
  beforeAll(() => {
    Element.prototype.scrollIntoView = jest.fn();
    Element.prototype.scrollTo = jest.fn();

    Object.defineProperty(
      globalThis as unknown as { ResizeObserver?: jest.Mock },
      'ResizeObserver',
      {
        writable: true,
        configurable: true,
        value: jest.fn(() => ({
          observe: jest.fn(),
          unobserve: jest.fn(),
          disconnect: jest.fn(),
        })),
      }
    );
  });

  afterAll(() => {
    jest.restoreAllMocks();
    delete (globalThis as unknown as { ResizeObserver?: jest.Mock })
      .ResizeObserver;
  });

  it('Should Primary mode render successfully', () => {
    const { baseElement } = render(
      <ScrollableHorizontal
        description="Discover our global impact and collective support for communities."
        heading="Impact Summary"
        itemsLength={1}
        mode="primary"
        name="child"
        showDescription
      >
        <DataCard
          content="Our livelihood improvement programs enable millions globally to reach better outcomes"
          heading="4M+"
          logoName="food"
          logoText="Food"
          size="1"
          subHeading="People served"
          variant="surface"
        />
      </ScrollableHorizontal>
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('Should Inverse mode render successfully', () => {
    const { baseElement } = render(
      <ScrollableHorizontal
        description="Discover our global impact and collective support for communities."
        heading="Impact Summary"
        itemsLength={1}
        mode="inverse"
        name="test"
        showDescription
      >
        <DataCard
          content="Our livelihood improvement programs enable millions globally to reach better outcomes"
          heading="4M+"
          logoName="water"
          logoText="Water"
          size="1"
          subHeading="People served"
          variant="default"
        />
      </ScrollableHorizontal>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
