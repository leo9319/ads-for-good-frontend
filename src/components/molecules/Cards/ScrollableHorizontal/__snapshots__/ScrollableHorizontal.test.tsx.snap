// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scrollable Horizontal Component Should Inverse mode render successfully 1`] = `
<body>
  <div>
    <div
      class="rt-Box container inverse"
    >
      <div
        class="rt-Box header"
      >
        <div
          class="rt-Flex rt-r-fd-column contentHeader"
        >
          <p
            class="rt-Text mocked-class-name typography-font-family-default heading"
            id=""
          >
            Impact Summary
          </p>
          <p
            class="rt-Text mocked-class-name typography-font-family-default description"
            id=""
          >
            Discover our global impact and collective support for communities.
          </p>
        </div>
      </div>
      <div
        class="root body"
        dir="ltr"
        style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
      >
        <style>
          [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
        </style>
        <div
          class="viewport"
          data-radix-scroll-area-viewport=""
          style="overflow-x: scroll; overflow-y: hidden;"
          tabindex="0"
        >
          <div
            style="min-width: 100%; display: table;"
          >
            <div
              class="rt-Flex contentContainer"
            >
              <div
                class="rt-reset rt-BaseCard rt-Card rt-r-size-1 rt-variant-surface dataCardContainer size-1 default"
              >
                <div
                  class="rt-Box headerContainer"
                >
                  <div
                    class="rt-Box header"
                  >
                    <p
                      class="rt-Text mocked-class-name typography-font-family-default heading"
                      id=""
                    >
                      4M+
                    </p>
                    <p
                      class="rt-Text mocked-class-name typography-font-family-default subHeading"
                      id=""
                    >
                      People served
                    </p>
                  </div>
                  <p
                    class="rt-Text mocked-class-name typography-font-family-default content"
                    id=""
                  >
                    Our livelihood improvement programs enable millions globally to reach better outcomes
                  </p>
                </div>
                <div
                  class="rt-Box footer"
                >
                  <div
                    class="tagComp"
                    style="pointer-events: none;"
                  >
                    <span
                      class="rt-reset rt-Badge rt-r-size-1 rt-variant-soft default"
                      data-accent-color=""
                      style="padding-inline: 4px 12px; border-radius: 20px; cursor: unset; width: auto; max-width: 100%; height: 32px; padding-block: 0;"
                      tabindex="0"
                    >
                      <span
                        class="rt-reset rt-AvatarRoot rt-r-size-1 rt-variant-soft"
                        data-accent-color=""
                        data-radius="full"
                      >
                        <span
                          class="rt-AvatarFallback"
                        />
                      </span>
                      <span
                        class="rt-Text rt-truncate mocked-class-name typography-font-family-default tagText"
                        id=""
                      >
                        Water
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Scrollable Horizontal Component Should Primary mode render successfully 1`] = `
<body>
  <div>
    <div
      class="rt-Box container primary"
    >
      <div
        class="rt-Box header"
      >
        <div
          class="rt-Flex rt-r-fd-column contentHeader"
        >
          <p
            class="rt-Text mocked-class-name typography-font-family-default heading"
            id=""
          >
            Impact Summary
          </p>
          <p
            class="rt-Text mocked-class-name typography-font-family-default description"
            id=""
          >
            Discover our global impact and collective support for communities.
          </p>
        </div>
      </div>
      <div
        class="root body"
        dir="ltr"
        style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
      >
        <style>
          [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
        </style>
        <div
          class="viewport"
          data-radix-scroll-area-viewport=""
          style="overflow-x: scroll; overflow-y: hidden;"
          tabindex="0"
        >
          <div
            style="min-width: 100%; display: table;"
          >
            <div
              class="rt-Flex contentContainer"
            >
              <div
                class="rt-reset rt-BaseCard rt-Card rt-r-size-1 rt-variant-surface dataCardContainer size-1 surface"
              >
                <div
                  class="rt-Box headerContainer"
                >
                  <div
                    class="rt-Box header"
                  >
                    <p
                      class="rt-Text mocked-class-name typography-font-family-default heading"
                      id=""
                    >
                      4M+
                    </p>
                    <p
                      class="rt-Text mocked-class-name typography-font-family-default subHeading"
                      id=""
                    >
                      People served
                    </p>
                  </div>
                  <p
                    class="rt-Text mocked-class-name typography-font-family-default content"
                    id=""
                  >
                    Our livelihood improvement programs enable millions globally to reach better outcomes
                  </p>
                </div>
                <div
                  class="rt-Box footer"
                >
                  <div
                    class="tagComp"
                    style="pointer-events: none;"
                  >
                    <span
                      class="rt-reset rt-Badge rt-r-size-1 rt-variant-soft default"
                      data-accent-color=""
                      style="padding-inline: 4px 12px; border-radius: 20px; cursor: unset; width: auto; max-width: 100%; height: 32px; padding-block: 0;"
                      tabindex="0"
                    >
                      <span
                        class="rt-reset rt-AvatarRoot rt-r-size-1 rt-variant-soft"
                        data-accent-color=""
                        data-radius="full"
                      >
                        <span
                          class="rt-AvatarFallback"
                        />
                      </span>
                      <span
                        class="rt-Text rt-truncate mocked-class-name typography-font-family-default tagText"
                        id=""
                      >
                        Food
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
